CRM/Contact/Import/Parser/Contact.php

```diff
@@ -262,10 +262,42 @@ public function init() {
   }
 
   /**
-   * Get configured contact type.
+   * Gets the fields available for importing in a key-name, title format.
+   *
+   * @return array
+   *   eg. ['first_name' => 'First Name'.....]
+   *
+   * @throws \API_Exception
+   *
+   * @todo - we are constructing the metadata before we
+   * have set the contact type so we re-do it here.
+   *
+   * Once we have cleaned up the way the mapper is handled
+   * we can ditch all the existing _construct parameters in favour
+   * of just the userJobID - there are current open PRs towards this end.
    */
-  protected function getContactType() {
-    return $this->_contactType ?? 'Individual';
+  public function getAvailableFields(): array {
+    $this->setFieldMetadata();
```

Comment:
`setFieldMetadata` means we don't need to call `run` or `init` first - but I think this will evolve as we clean up the inputs on construct